### PR TITLE
Add log bench for cpp api

### PR DESCRIPTION
**Pull request by alex391a**
* Add log bench for cpp api in runKernelFromInvocation

### CODE REVIEW

**Review by AndySu12 — DISMISSED** [review]

**Comment by KKyang** [comment]
Can we move the log for C API to tensile_host.cpp and share the same function?

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 1]
```diff
@@ -1821,6 +1821,59 @@ rocblaslt_status runKernelFromInvocation(rocblaslt_handle       handle,
...
+                data->problem.alphaRestriction(),
+                "--beta",
+                data->problem.betaRestriction(),
+                "--transA",
+                data->problem.transA() ? "T" : "N",
+                "--transB",
+                data->problem.transB() ? "T" : "N",
+                "--batch_count",
+                data->problem.batchSize(0),
+                (gemmType == rocblaslt::RocGemmType::ROCBLASLT_GROUPED_GEMM) ? "--grouped_gemm" : "",
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp]: this code section is in (gemmType == rocblaslt::RocGemmType::ROCBLASLT_GEMM)

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 2]
```diff
@@ -1821,6 +1821,59 @@ rocblaslt_status runKernelFromInvocation(rocblaslt_handle       handle,
...
+                GEN_BENCH_ARG(hipDataType_to_bench_string, "--a_type", tensile2HipType(data->problem.a().dataType())),
+                GEN_BENCH_ARG(hipDataType_to_bench_string, "--b_type", tensile2HipType(data->problem.b().dataType())),
+                GEN_BENCH_ARG(hipDataType_to_bench_string, "--c_type", tensile2HipType(data->problem.c().dataType())),
+                GEN_BENCH_ARG(hipDataType_to_bench_string, "--d_type", tensile2HipType(data->problem.d().dataType())),
+                GEN_BENCH_ARG(hipDataType_to_bench_string, "--compute_type", tensile2HipType(data->problem.computeType())),
+                GEN_BENCH_ARG(hipDataType_to_bench_string, "--scale_type", tensile2HipType(data->problem.alphaType())),
+                GEN_BENCH_ARG(hipDataType_to_bench_string, "--bias_type", tensile2HipType(data->problem.bias().dataType()))
+                );
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp]: where are scaleA/B/C/D, scale Alpha Vector and other epilogues?

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 3]
```diff
@@ -1821,6 +1821,59 @@ rocblaslt_status runKernelFromInvocation(rocblaslt_handle       handle,
...
+ 
+#undef GEN_BENCH_ARG
+            }
             status = hip2RocStatus(adapter->launchKernels(data->kernels, stream, start, stop));
         }
         else if(gemmType == rocblaslt::RocGemmType::ROCBLASLT_GROUPED_GEMM)
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp:1891]: group gemm also need log_bench

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 4]
```diff
@@ -1821,6 +1821,59 @@ rocblaslt_status runKernelFromInvocation(rocblaslt_handle       handle,
...
+                "--stride_a",
+                data->problem.a().strides()[2],
+                "--stride_b",
+                data->problem.b().strides()[2],
+                "--stride_c",
+                data->problem.c().strides()[2],
+                "--stride_d",
+                data->problem.d().strides()[2],
+                "--alpha",
+                data->problem.alphaRestriction(),
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp]: should be in data->inputs

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 5]
```diff
@@ -333,6 +333,68 @@ namespace
...
+            problem.useScaleCD() ? "--scaleD" : "",
+            problem.useScaleAlphaVec() ? "--scaleAlpha_vector" : "",
+            GEN_BENCH_ARG(hipDataType_to_bench_string, "--a_type", tensile2HipType(problem.a().dataType())),
+            GEN_BENCH_ARG(hipDataType_to_bench_string, "--b_type", tensile2HipType(problem.b().dataType())),
+            GEN_BENCH_ARG(hipDataType_to_bench_string, "--c_type", tensile2HipType(problem.c().dataType())),
+            GEN_BENCH_ARG(hipDataType_to_bench_string, "--d_type", tensile2HipType(problem.d().dataType())),
+            GEN_BENCH_ARG(hipDataType_to_bench_string, "--compute_type", tensile2HipType(problem.computeType())),
+            GEN_BENCH_ARG(hipDataType_to_bench_string, "--scale_type", tensile2HipType(problem.alphaType())),
+            GEN_BENCH_ARG(hipDataType_to_bench_string, "--bias_type", tensile2HipType(problem.bias().dataType())));
+            // TODO: Tensile::ActivationType (to rocblaslt_epilogue) to bench string?
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp:392]: if not ready, I would suggest label this PR as "Draft" or "noCI"

**Pull request closed without merging**